### ISSUE TITLE
fix(Generator): run Docker as non-root user

### ISSUE THREAD

**Issue by nokome** [stencila/dockter#25]
It's best practice to run processes in Docker containers as a [non-root user](https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b)

So we should add something like the following to generated Docker files.
```
RUN useradd --create-home --uid 1001 -s /bin/bash user
USER user
WORKDIR /home/user
```

**Comment by apawlik** [comment]
This should address it https://github.com/stencila/dockter/commit/07f8c4b6bacfded7acec6bea7ee3b87f12b1d1b0

However, I am wondering if
a) the Dockerfile should actually grab the UID for the user currently running dockter
b) using UID way above 1001 could be a better idea since 1001 may actually be already assigned (http://www.linfo.org/uid.html)

**Comment by nokome** [comment]
Thanks! I made a couple of fixes in https://github.com/stencila/dockter/commit/766ad6e2e34fe511904d40b708a227bd46e60f39:

- removed the `\\` line continuation chars since these are three separate, single-line Dockerfile instructions and they won't work if you put them on a single line
- added the additional block of Dockerfile code to the test expectations so that the tests pass.

Re points above:

(a) I think that is a good idea for when we `execute` an image by using the `--user` option (this is a common pattern that I saw Finlay use yesterday as it happens). But because someone else might use run build the generated Dockerfile or run the image elsewhere it seems inappropriate/uneccesary to use the uid of the current user when generating the Dockerfile

(b) Remember that to get a UID clash someone would have to do `useradd` in one of the base images we use e.g. `ubuntu:18.04`, `python:2.7`. So the risk is low and we'll no about it very quickly when running the unit tests. But I've got no problem changing it to something more random e.g. `4124`.

A friendly reminder to run `make lint test` before committing and to use [changelog style commit messages](https://github.com/stencila/dockter/blob/master/CONTRIBUTING.md#commit-messages) for better semantic versioning and release notes.
